include/Dialect/Polynomial/IR/PolynomialTypes.td

```diff
@@ -4,16 +4,17 @@
 include "include/Dialect/Polynomial/IR/PolynomialDialect.td"
 include "include/Dialect/Polynomial/IR/PolynomialAttributes.td"
 
-include "mlir/IR/DialectBase.td"
 include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/BuiltinTypes.td"
+include "mlir/IR/DialectBase.td"
 
 // A base class for all types in this dialect
-class Polynomial_Type<string name, string typeMnemonic>
-    : TypeDef<Polynomial_Dialect, name> {
+class Polynomial_Type<string name, string typeMnemonic, list<Trait> traits = []>
```

Comment:
Adding a pass-through list of traits is, imho, a good pattern to follow for any tablegen type `class` we define, so we should consider just adding this to all of them by default.